enhancements/update/targeted-update-edge-blocking.md

```diff
@@ -0,0 +1,400 @@
+
+Older update services consuming newer graph-data will know that they are missing some features unique to 1.1.0, but that they will still get something reasonable out of the data by using their 1.0.0 parser (they'll just consider all conditional edges to be complete blockers).
+
+Newer clients talking to older update services will not receive any `conditionalEdges`, but they will understand all the data that the update service sends to them.
+Older clients talking to newer update services will not notice `conditionalEdges`, so those edges will continue to be unconditionally blocked for those clients.
+
+Newer clients consuming older ClusterVersion will not receive any `conditionalUpdates`, but they will understand all the data included in the ClusterVersion object (e.g. `availableUpdates`).
+Older clients consuming newer ClusterVersion will not notice `conditionalUpdaets`, so those edges will continue to be unconditionally blocked for those clients.
+
+## Implementation History
+
+Major milestones in the life cycle of a proposal should be tracked in `Implementation
+History.
+
+## Drawbacks
+
+Dynamic edge status that is dependent on cluster state makes [the graph-data repository][graph-data] a less authoritative view of the graph served to a given client at a given time, as discussed in [the *risks and mititgations* section](#risks-and-mitigations).
+This is mitigated by ClusterVersion's [`status.history[].overrides'](#enhanced-clusterversion-representation), which records any cluster-version operator objections which the cluster administrator chose to override.
+It is possible that cluster administrators would chose to clear that data, but it seems unlikely that they would invest significant effort in trying to cover their tracks when [the edges are supported regardless of whether they were recommended][openshift-docs-32091].
+
+## Alternatives
+
+### Additional data sources
+
+The update service could switch on data scraped from Insights tarballs or other sources instead of Prometheus.
+And we could extend `clusters` in future work to allow for that.
+With this initial enhancement, I focused on Prometheus because it already exposes an API and provides access to lots of in-cluster data via a single [PromQL][] query string.
+
+#### Update-service-side filtering
+
+Instead of filtering cluster-side in the cluster-version operator, we could filter edges on the update-service side by querying [uploaded Telemetry][uploaded-telemetry] or [client-provided query parameters][cincinnati-for-openshift-request].
+However, there is more data available in-cluster beyond what is uploaded as Telemetry.
+And because we are [supporting edges which we do not recommend][openshift-docs-32091], we'd need to pass the reasons for not recommending those edges out to clusters anyway.
+Passing enough information to make the decision completely on the cluster side is not that much more work.
```

Comment:
In on-prem OSUS scenarios they'd need a telemetry sink and Prometheus service to poll, does such exist via OSUS today?

Reply:
not having a local Telemetry target makes the service-side PromQL filtering (#427) more difficult, but service-side filtering based on client-provided query parameters (#737) would work fine.  But for me, the main reason to shift to cluster-side processing is that the conditional edges are still supported, regardless of whether they are recommended, so the viability of service-side filtering seems somewhat academic.

Reply:
Doing the evaluation on the cluster also makes it possible to have it work for disconnected environments.